### PR TITLE
Log headers from unexpected response

### PR DESCRIPTION
**Pull request by griffitr**
We're receiving 504 connection errors and need further, temporary insight into which layer is returning them.

### CODE REVIEW

File: lib/new_relic/harvest/collector/protocol.ex [hunk 1]
```diff
@@ -180,6 +180,23 @@ defmodule NewRelic.Harvest.Collector.Protocol do
...
   end
 
+  defp log_error(status, error, params, body, headers) do
+    case Jason.decode(body) do
+      {:ok, %{"exception" => exception}} ->
+        NewRelic.log(
+          :error,
+          "#{params[:method]}: (#{status}) #{error} - " <>
+            "#{exception["error_type"]} - #{exception["message"]} - " <>
+            "headers: [#{Enum.map(headers, fn {key, value} -> "#{key}: #{value} " end)}]"
```
Comment (mattbaker) [inline lib/new_relic/harvest/collector/protocol.ex]: Could also just say `"headers: #{inspect(headers)}"`

Comment (mattbaker) [reply lib/new_relic/harvest/collector/protocol.ex]: Ditto L195

**Comment by mattbaker** [comment]
🐑 

Did you manage to test it out just to confirm it behaves the way we expect? 

Could also test it out in the elixir dummy or something if it's easier

**Comment by griffitr** [comment]
@mattbaker I ran the existing test suite and saw the output in the logs there.

**Pull request merged**